lib/machine_gun.ex

```diff
@@ -259,8 +259,13 @@ defmodule MachineGun do
       worker = :poolboy.checkout(pool, pool_queue, pool_timeout)
 
       try do
-        with {:ok, response} <- Worker.request(worker, request, request_timeout, m_mod, m_state) do
-          {:ok, %Response{response | request_url: url}}
+        if worker == :full do
+          {:error, %Error{reason: :pool_full}}
```

Comment:
this seems an event we'd want to metric if we are not already doing it. Not here obviously but where we receive this response. This could become an alarm potentially.